src/App.tsx

```diff
@@ -21,75 +21,126 @@ enum SortType {
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort(
+    (
+      wordOne, wordTwo,
+    ) => getSortChoice(
+      wordOne, wordTwo, sortType,
+    ),
+  );
+
+  if (isReversed === true) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component <{}, State> {
+  state = {
+    isReversed: false,
+    sortType: 0,
+  };
+
+  sortByAlphabet = () => {
+    this.setState({ isReversed: false, sortType: SortType.ALPHABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ isReversed: false, sortType: SortType.LENGTH });
+  };
+
+  reverse = () => {
+    this.setState(statePrev => ({
+      isReversed: !statePrev.isReversed, sortType: statePrev.sortType,
+    }));
+  };
+
+  reset = () => {
+    this.setState({ isReversed: false, sortType: SortType.NONE });
+  };
+
+  render() {
+    const sort = this.state.sortType;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={sort === 1
+              ? 'button is-info' : 'button is-info is-light'}
+            onClick={this.sortByAlphabet}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={sort === 2
+              ? 'button is-success' : 'button is-success is-light'}
+            onClick={this.sortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={this.state.isReversed === true
+              ? 'button is-warning' : 'button is-warning is-light'}
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+
+          {this.state.sortType !== 0 || this.state.isReversed === true ? (
```

Comment:
use https://react.dev/learn/conditional-rendering instead of ternary in such cases

Reply:
everything is fixed )